lib/api/BackbeatAPI.js

```diff
@@ -501,6 +501,50 @@ class BackbeatAPI {
         });
     }
 
+    /**
+     * Decrement any existing failed metrics to account for the retry.
+     * @param {String} site - The site of the object
+     * @param {Number} contentLength - The size of the object
+     * @param {Function} cb - The callback to call
+     * @return {undefined}
+     */
+    _decrementFailedMetrics(site, contentLength, cb) {
+        const opsFailQS = `${site}:${redisKeys.opsFail}:*`;
+        const bytesFailQS = `${site}:${redisKeys.bytesFail}:*`;
+        async.series([
+            // Decrement ops failed count by one
+            next => this._redis.scan(opsFailQS, undefined, (err, keys) => {
+                if (keys.length === 0) {
+                    return next();
+                }
+                const key = keys[keys.length - 1];
```

Comment:
I don't think order is guaranteed by the SCAN command, it's better to sort on our end to pick the latest key.

Reply:
Was thinking similarly. However I actually realized some additional complexity here: even if decrementing the latest key, the failure could have happened > 20 min ago (beyond the expiry time) so I will have to be sensitive to this fact or else the metrics could go into the negative.

Reply:
I'm planning to add a normalized timestamp to the failure key so that I can then match the key here, or else not decrement the failures metric.

Reply:
Gotcha. Also, duplicates. SCAN can return duplicates.